test/controllers/survey_controller_test.exs

```diff
@@ -616,12 +701,24 @@ defmodule Ask.SurveyControllerTest do
        mobile_contact_messages: mobile_contact_messages}
     end
 
+    test "limits endpoint for survey simulations only", %{
+      create_running_survey: create_running_survey,
+      get_simulation_initial_state: get_simulation_initial_state
+    } do
+      [mode, simulation] = ["sms", false]
+      %{survey: survey} = create_running_survey.(mode, simulation)
+
+      assert_error_sent(:not_found, fn ->
+        get_simulation_initial_state.(survey, mode)
+      end)
+    end
+
     test "SMS return an empty map", %{
       create_running_survey: create_running_survey,
       get_simulation_initial_state: get_simulation_initial_state
     } do
-      mode = "sms"
-      %{survey: survey} = create_running_survey.(mode)
+      [mode, simulation] = ["sms", true]
```

Comment:
I may be a little rusted with elixir, but this is the same as
```elixir
mode = "sms"
simulation = true
```

right?

Reply:
Yes, I think it's exactly the same. It's just a matter of style. May I be missing some details regarding performance, or something else?

Reply:
No that I can see 
I was just checking that was the same 🙂

Reply:
[Here](https://elixir-lang.org/getting-started/pattern-matching.html#pattern-matching) there are some similar examples:

```elixir
iex> {a, b, c} = {:hello, "world", 42}
{:hello, "world", 42}
iex> a
:hello
iex> b
"world"
```

```elixir
iex> [a, b, c] = [1, 2, 3]
[1, 2, 3]
iex> a
1
```

Maybe using a tuple (instead of a list) is more expressive?

Reply:
for me the simplest way is to bind the two variables separately, but choose the one that best suits your taste 🙂

Reply:
So, after taking the advice of my gut (and, ok, also of @bcardiff 😬) tuples they are 🎉